File: py-basics/src/lectures/dictionaries/leap-years.py
# 7kyu | Leap Years
# https://www.codewars.com/kata/526c7363236867513f0005ca/python


def isLeapYear(year):
    if year % 400 == 0:
        return True
    elif year % 100 == 0:
        return False
    elif year % 4 == 0:
        return True
    else:
        return False


# Leap years should return True
print(isLeapYear(1984))  # Year 1984 was a leap year
print(isLeapYear(2000))  # Year 2000 was a leap year!

# Leap years should return False
print(isLeapYear(1234))  # Year 1234 was NOT a leap year!
print(isLeapYear(1100))  # Year 1100 was NOT a leap year!

File: js-basics/src/lectures/variables/leap-years.js
/**
 * Lektion: Abschluss
 * 7kyu | Leap Years
 * https://www.codewars.com/kata/526c7363236867513f0005ca/javascript
 */

function isLeapYear(year) {
    if (year % 400 === 0) return true;
    if (year % 100 === 0) return false;
    if (year % 4 === 0) return true;
    return false;
}

console.log(isLeapYear(1234)); // false
console.log(isLeapYear(1984)); // true
console.log(isLeapYear(2000)); // true
console.log(isLeapYear(2010)); // false
console.log(isLeapYear(2013)); // false